mistletoe/block_tokenizer.py

```diff
@@ -78,7 +89,19 @@ def tokenize_block(iterable, token_types, start_line=1):
     line = lines.peek()
     while line is not None:
         for token_type in token_types:
-            if token_type.start(line):
+            # We need to support two cases:
+            # * a start() method accepting only a single `line` parameter
+            # * a start() method accepting both a `line` and a `lines` parameter, e.g. DefinitionList
+            func_args_count = token_type.start_args_count()
+            if func_args_count == 1:
+                is_start = token_type.start(line)
+            elif func_args_count == 2:
+                is_start = token_type.start(line, lines.reader())
+            else:
+                raise NotImplementedError(
+                    f"start() method of {token_type} has an incorrect number of parameters: {func_args_count}"
```

Comment:
Note: While we still support Python 3.5 (pretty old, right? :)), we need to rewrite this to the old style:

```suggestion
                    "start() method of {} has an incorrect number of parameters: {}".format(token_type, func_args_count)
```